src/main.c

```diff
@@ -287,6 +289,32 @@ static void callback(const nrsc5_event_t *evt, void *opaque)
                 log_info("XHDR: %d %08X %d", evt->id3.xhdr.param, evt->id3.xhdr.mime, evt->id3.xhdr.lot);
         }
         break;
+    case NRSC5_EVENT_SIG:
+        sig_service = evt->sig.services;
+        while (sig_service)
```

Comment:
You could use a for-loop instead of the while loop to reduce the code a bit. Same with the inner loop as well.

Reply:
Done.